view/src/__tests__/DiffViewWrapper.test.jsx

```diff
@@ -0,0 +1,325 @@
+// Unit tests of DiffViewWrapper component
+// Author: Lukas Petr
+
+import {
+  cleanup, render, screen, within,
+} from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+
+import DiffViewWrapper from '../components/DiffViewWrapper';
+
+/**
+ * Helper function to test if shown line number and line of code match
+ * to original line numbering.
+ * @param {string} oldCode Old source of code.
+ * @param {string} newCode New source of code.
+ * @param {Function} oldLineCallback Function which is called for each line of
+ * old code, function receives shown line number and shown line of code.
+ * @param {Function} newLineCallback Same as oldLineCallback but for new code.
+ */
+function testCodeNumberMatch(
+  oldCode,
+  newCode,
+  oldLineCallback,
+  newLineCallback,
+) {
+  const oldCodeLines = oldCode.split('\n');
+  const newCodeLines = newCode.split('\n');
+  const rows = screen.getAllByRole('row');
+  rows.forEach((row) => {
+    const cells = within(row).getAllByRole('cell');
+    // is row with code
+    if (cells.length !== 4) {
+      return;
+    }
+    // old lines check
+    if (cells[0].textContent !== '') {
+      const lineNumber = Number(cells[0].textContent);
+      const lineCode = cells[1].textContent;
+      try {
+        expect(lineCode).toBe(oldCodeLines[lineNumber - 1]);
+      } catch (e) {
+        e.message += `\nReceived line number in old file: ${lineNumber}`;
+        throw e;
+      }
+      oldLineCallback(lineNumber, lineCode);
+    }
+    // new lines check
+    if (cells[2].textContent !== '') {
+      const lineNumber = Number(cells[2].textContent);
+      const lineCode = cells[3].textContent;
+      try {
+        expect(lineCode).toBe(newCodeLines[lineNumber - 1]);
+      } catch (e) {
+        e.message += `\nReceived line number in new file: ${lineNumber}`;
+        throw e;
+      }
+      newLineCallback(lineNumber, lineCode);
+    }
+  });
+}
+
+describe('testing a differing function visualisation', () => {
+  /* eslint-disable no-tabs */
+  const diff = `--- /tmp/tmp47pt0hsi/1	2023-03-16 17:36:00.611068650 +0100
++++ /tmp/tmp47pt0hsi/2	2023-03-16 17:36:00.611068650 +0100
+@@ -770,4 +750,6 @@
+ 
+-	if (!(flags & DEQUEUE_SAVE))
++	if (!(flags & DEQUEUE_SAVE)) {
+ 		sched_info_dequeued(rq, p);
++		psi_dequeue(p, flags & DEQUEUE_SLEEP);
++	}
+`;
+  /* eslint-enable no-tabs */
+
+  /* eslint-disable no-tabs */
+  const oldCode = `${'\n'.repeat(765)
```

Comment:
Compute the number from the `oldStart` var below:
```suggestion
  const oldCode = `${'\n'.repeat(oldStart - 1)
```